python/ray/data/datasource/file_metadata_shuffler.py

```diff
@@ -1,34 +1,33 @@
-from typing import Any, Dict, List, Tuple
+import sys
+from typing import Any, List, Union
 
+import numpy as np
 
-class FileMetadataShuffler:
-    """Abstract class for file metadata shuffler.
-
-    Shufflers live on the driver side of the Dataset only.
-    """
+if sys.version_info >= (3, 8):
+    from typing import Literal
+else:
+    from typing_extensions import Literal
 
-    def __init__(self, reader_args: Dict[str, Any]):
-        self._reader_args = reader_args
 
-    def shuffle_files(
-        self,
-        paths_and_sizes: List[Tuple[str, int]],
-    ) -> List[Tuple[str, int]]:
-        """Shuffle files in the given paths and sizes.
-
-        Args:
-            paths_and_sizes: The file paths and file sizes to shuffle.
-
-        Returns:
-            The file paths and their sizes after shuffling.
-        """
-        raise NotImplementedError
+class FileMetadataShuffler:
```

Comment:
IMO this abstractions adds an unnecessary layer on indirection. It essential just wraps a single function. I think it'd be simpler if did something like this in `FileBasedDatasource`:

```python
if shuffle == "files":
    metadata = np.random.shuffle(metadata)
```

If we introduce different shuffling methods in the future, we can always revisit this an introduce a new abstraction. But at this point, I think it's premature

Reply:
Looks like the `FileMetadataShuffler` class is still here (?)